Tests/RegexTests/RegexDSLTests.swift

```diff
@@ -118,22 +143,47 @@ class RegexDSLTests: XCTestCase {
       ("t" | "k").capture() // Substring
       oneOf { "k".capture(); "j".capture() } // (Substring?, Substring?)
     }
-    // Assert the inferred capture type.
-    let _: (Substring, Substring, Substring, [Substring], Substring, Substring?, Substring?).Type
-      = type(of: regex).Match.self
-    let maybeMatch = "aaaabccccdddkj".match(regex)
-    let match = try XCTUnwrap(maybeMatch).match
-    XCTAssertEqual(match.0, "aaaabccccdddkj")
-    XCTAssertEqual(match.1, "b")
-    XCTAssertEqual(match.2, "cccc")
-    XCTAssertEqual(match.3, ["d", "d", "d"])
-    XCTAssertEqual(match.4, "k")
-    XCTAssertEqual(match.5, .none)
-    XCTAssertEqual(match.6, .some("j"))
+  }
+  
+  func testQuantificationBehavior() throws {
+    try _testDSLCaptures(
+      ("abc1def2", ("abc1def2", ["2"])),
+      captureType: (Substring, [Substring]).self, ==)
+    {
+      oneOrMore {
+        oneOrMore(.word)
+        CharacterClass.digit.capture()
+      }
+    }
+
+    try _testDSLCaptures(
+      ("abc1def2", ("abc1def2", ["1", "2"])),
+      captureType: (Substring, [Substring]).self, ==)
+    {
+      oneOrMore {
+        oneOrMore(.word, .reluctantly)
+        CharacterClass.digit.capture()
```

Comment:
Can we drop `CharacterClass`?

At some point I had to add a RegexWithComponent protocol and some overloads to make the static vars discoverable, but @rxwei was able to remove that and keep the lookup. I don't recall the issues there.

Reply:
Is the compiler able to distinguish between `.digit` being a method of `oneOrMore(...)`'s return type and it being a static member call on its own? I haven't been able to get that kind of lookup to work for static protocol methods.